### ISSUE TITLE
ansible-pull: Shallow clone broken in 2.1.0.0-1

### ISSUE THREAD

**Issue by rodecker** [ansible/ansible-modules-core#3782]
##### ISSUE TYPE
- Bug Report
##### COMPONENT NAME

ansible-pull
##### ANSIBLE VERSION

```
ansible 2.1.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

None
##### OS / ENVIRONMENT

Ubuntu 12.04 (Precise)
##### SUMMARY

After upgrading to 2.1.0.0-1 ansible-pull no longer sees changes when doing a shallow clone.
Running ansible-pull with --full still works.
##### STEPS TO REPRODUCE

```
ansible-pull -d /etc/ansible/ring -U https://github.com/NLNOG/ring-ansible.git -i nodes -l <hostname> -c local playbook.yml
```
##### EXPECTED RESULTS

New changes are pulled.
##### ACTUAL RESULTS

Recent changes in the repository are not seen.

```
# ansible-pull -vvv -d /etc/ansible/ring -U https://github.com/NLNOG/ring-ansible.git -i nodes -l <hostname> -c local playbook.yml 
Using /etc/ansible/ansible.cfg as config file
Starting Ansible Pull at 2016-05-29 08:38:59
/usr/bin/ansible-pull -vvv -d /etc/ansible/ring -U https://github.com/NLNOG/ring-ansible.git -i nodes -l <hostname> -c local playbook.yml
Using /etc/ansible/ansible.cfg as config file
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: root
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1464511140.14-77534406207186 `" && echo ansible-tmp-1464511140.14-77534406207186="` echo $HOME/.ansible/tmp/ansible-tmp-1464511140.14-77534406207186 `" ) && sleep 0'
<localhost> PUT /tmp/tmpOAZtUz TO /root/.ansible/tmp/ansible-tmp-1464511140.14-77534406207186/git
<localhost> EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /root/.ansible/tmp/ansible-tmp-1464511140.14-77534406207186/git; rm -rf "/root/.ansible/tmp/ansible-tmp-1464511140.14-77534406207186/" > /dev/null 2>&1 && sleep 0'
localhost | SUCCESS => {
    "after": "159e4f9cf7f8f41915713360d9c66e5daecdd7f9", 
    "before": "159e4f9cf7f8f41915713360d9c66e5daecdd7f9", 
    "changed": false, 
    "invocation": {
        "module_args": {
            "accept_hostkey": false, 
            "bare": false, 
            "clone": true, 
            "depth": 1, 
            "dest": "/etc/ansible/ring", 
            "executable": null, 
            "force": false, 
            "key_file": null, 
            "name": "https://github.com/NLNOG/ring-ansible.git", 
            "recursive": true, 
            "reference": null, 
            "refspec": null, 
            "remote": "origin", 
            "repo": "https://github.com/NLNOG/ring-ansible.git", 
            "ssh_opts": null, 
            "track_submodules": false, 
            "update": true, 
            "verify_commit": false, 
            "version": "HEAD"
        }, 
        "module_name": "git"
    }
}
```

**Comment by robinro** [comment]
I believe f1bacc1d3f9578f26d4ae2f66112cbb2509a7fe8 broke this. I'll have a look.

**Comment by robinro** [comment]
@rodecker Thanks for the report.
The case without version specified is something I obviously didn't test thoroughly enough when working on shallow clones.

I pushed a fix and a test for the git module. Please have a look and let me know whether this fixes your issue.

Another workaround would be to specify the version argument `-C master`. This is what we use in production and also what the tests covered so far.

**Comment by rodecker** [comment]
Thanks for looking into this.

The commit does not resolve the issue for me. The `-C master` workaround also does not pick up the changes.

**Comment by robinro** [comment]
Does the test in https://github.com/ansible/ansible/pull/16055 pass for you?

Can you post a minimal example (similar to the test) that fails for you?

**Comment by rodecker** [comment]
Found the problem.

Ubuntu 12.04 has git 1.7.9.5 installed, on which (as the documentation of git.py states) the depth parameter does not function correctly. Before https://github.com/ansible/ansible-modules-core/commit/f1bacc1d3f9578f26d4ae2f66112cbb2509a7fe8 depth was not used, so ansible-pull worked as expected.

I checked on Ubuntu 14.04 (git 1.9.1). The problem does not exist there.

**Comment by robinro** [comment]
I updated the PR to check for the git version and fall back to a full clone on too old git versions.

**Comment by rodecker** [comment]
This works. Thanks!

**Comment by abadger** [comment]
The update to the git module broke integration tests.  Revert those for now so reopening this until that's sorted out.

**Comment by robinro** [comment]
I cleaned up the PR:
https://github.com/ansible/ansible-modules-core/pull/3912 (fix)
https://github.com/ansible/ansible/pull/16224 (integration tests)

**Comment by robinro** [comment]
@rodecker This issue is fixed in the freshly released 2.1.1rc2.
